verde/tests/test_spline.py

```diff
@@ -99,6 +100,25 @@ def test_spline():
     )
 
 
+def test_spline_cv_scoring():
+    "Check scoring parameter works with SplineCV"
+    region = (100, 500, -800, -700)
+    synth = CheckerBoard(region=region)
+    data = synth.scatter(size=1500, random_state=1)
+    coords = (data.easting, data.northing)
+    # Compare SplineCV to results from Spline with cross_val_score
+    for score in ["r2", "neg_root_mean_squared_error"]:
+        spline = Spline()
+        score_spline = np.mean(
+            cross_val_score(spline, coords, data.scalars, scoring=score)
+        )
+        # Limit SplineCV to a single parameter set equal to Spline's defaults
+        spline_cv = SplineCV(mindists=[1e-5], dampings=[None], scoring=score)
+        spline_cv.fit(coords, data.scalars)
+        score_spline_cv = spline_cv.scores_.max()
+        npt.assert_allclose(score_spline, score_spline_cv, rtol=1e-5)
```

Comment:
Any particular reason to check the mean against the max? Would checking all scores individually work for this (like below)?

```suggestion
        score_spline = cross_val_score(spline, coords, data.scalars, scoring=score)
        # Limit SplineCV to a single parameter set equal to Spline's defaults
        spline_cv = SplineCV(mindists=[1e-5], dampings=[None], scoring=score)
        spline_cv.fit(coords, data.scalars)
        npt.assert_allclose(score_spline, spline_cv.scores_, rtol=1e-5)
```

Reply:
If I understand correctly, the `scores_` parameter of `SplineCV` contains the mean of the 5-fold CV scores for each combination of `mindists` and `dampings` i.e. the number of elements in `spline_cv.scores_` is equal to the number of parameter combinations (which is just one in this example). By contrast, `cross_val_score` returns the score for each fold (a 5-element array in this example).

So I think we do need to compare the mean from `cross_val_score` with the single result from `SplineCV`. However, I agree using `spline_cv.scores_.max()` in this case is confusing, because it's the "max" of a one-element array. Would

    score_spline_cv = spline_cv.scores_[0]

be clearer?

Reply:
Ah right, I forgot that `SplineCV` takes the mean. Yep, your suggestion is better and more explicit about what its doing 👍🏾